### PR TITLE
Switches default JDBC driver from MySQL to MariaDB

### PR DESCRIPTION
**Pull request by adriancole**
MySQL connector is GPL, even if it has an OSS exception.
https://www.mysql.com/about/legal/licensing/foss-exception/

MariaDB has a friendlier license, LGPL, which is less scary in audits.

### CODE REVIEW

**Comment by adriancole** [comment]
cc @mstine @abesto @dsyer @spencergibb

File: zipkin-java-server/src/main/resources/zipkin-server.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ server:
   port: ${QUERY_PORT:9411}
 spring:
   datasource:
-    driver-class-name: com.mysql.jdbc.Driver
+    driver-class-name: org.mariadb.jdbc.Driver
     url: jdbc:mysql://${mysql.host}:${mysql.port}/${mysql.db}?autoReconnect=true
```
Comment (adriancole) [inline zipkin-java-server/src/main/resources/zipkin-server.yml]: note: mariadb supports the same URI as the mysql driver

**Comment by abesto** [comment]
Are there any tech trade-offs we need to be aware of? (Disclaimer: I have zero experience with MariaDB)

**Comment by adriancole** [comment]
This is a different driver than the MySQL connector, and while we don't use extended flags there would be differences in how things like reconnect work. In both cases, these are drivers backed by a commercial company.

In the `zipkin-java-jdbc` project, unlike `zipkin-anormdb`, there's no tight coupling to a specific driver, so someone could swap it out to Oracle's driver. There's only one impact: self-tracing of JDBC is currently conditional on the driver type: We could probably move to a different condition, but regardless self-tracing of zipkin is more a showcase feature.

**Comment by abesto** [comment]
Alright, then this is a :+1: from me.

Let's make sure we get semver right when releasing; I guess minor would be the right call here.

**Comment by adriancole** [comment]
haha... this has never had a release! but yeah, sure thing on upstream

**Comment by abesto** [comment]
Oh right, we're on `zipkin-java`. Oops :)

**Comment by adriancole** [comment]
ex. #7 is incomplete.. definitely want to release this week

**Pull request merged**